internal/painter/image.go

```diff
@@ -163,3 +166,28 @@ func checkImageMinSize(img *canvas.Image, c fyne.Canvas, pixX, pixY int) bool {
 
 	return true
 }
+
+func isFileSVG(path string) bool {
+	return strings.ToLower(filepath.Ext(path)) == ".svg"
+}
+
+func isResourceSVG(res fyne.Resource) bool {
+	if strings.ToLower(filepath.Ext(res.Name())) == ".svg" {
+		return true
+	}
+
+	if len(res.Content()) < 5 {
+		return false
+	}
+
+	if strings.ToUpper(string(res.Content()[:5])) == "<!DOC" {
+		return true
+	}
+	if strings.ToLower(string(res.Content()[:5])) == "<?xml" {
+		return true
+	}
+	if strings.ToLower(string(res.Content()[:5])) == "<svg " {
+		return true
+	}
```

Comment:
Could be simplified;
```suggestion
	switch strings.ToUpper(string(res.Content()[:5])) {
		case "<!DOC", "<?xml", "<svg ":
			return true
	}
```